parser/internal/parservX/parser.go

```diff
@@ -887,6 +899,9 @@ func (p *parser) parseFunctionDefinition() (*ast.FunctionDefinition, error) {
 	if err != nil {
 		return nil, err
 	}
+	// Record the function parameter's name without the $ sign,
+	// because the logic to dereference a param does not use $ sign.
+	p.functionParameters[args[0].Name[1:]] = args[0]
```

Comment:
```suggestion
	p.functionParameters = args
```

and move the logic for removing the $ prefix to `parseFunctionArguments`?